rust/cardano-chain-follower/src/chain_sync.rs

```diff
@@ -411,14 +411,31 @@ async fn live_sync_backfill(
     Ok(())
 }
 
+/// Call the live sync backfill.
+/// This is a helper function to pause and resume the stats thread.
+async fn call_live_sync_backfill(
+    cfg: &ChainSyncConfig, update: &MithrilUpdateMessage,
+) -> anyhow::Result<()> {
+    stats::pause_thread(cfg.chain, stats::thread::name::LIVE_SYNC_BACKFILL_AND_PURGE);
+    let result = live_sync_backfill(cfg, update).await;
+    stats::resume_thread(cfg.chain, stats::thread::name::LIVE_SYNC_BACKFILL_AND_PURGE);
+    result
+}
+
 /// Backfill and Purge the live chain, based on the Mithril Sync updates.
 async fn live_sync_backfill_and_purge(
     cfg: ChainSyncConfig, mut rx: mpsc::Receiver<MithrilUpdateMessage>,
     mut sync_ready: SyncReadyWaiter,
 ) {
+    stats::start_thread(
```

Comment:
Move this to the spawn, something like:

```
 let _backfill_join_handle = spawn(async move {
        stats::start_thread(...);
        live_sync_backfill_and_purge(backfill_cfg.clone(), rx, sync_waiter).await;
        stats::stop_thread(...);        
    });
 ```

This task can end in many places, this will reliably catch that.
Its also not an infinitely running task, so it doesn't need pause/unpause.
Even though its waiting in certain places, its not reallly forever iterating. 
So remove that.

Its also not strictly a thread, its a tokio async task, however we can still use the start_thread/stop_thread tracking on it.